lib/jsdom/living/nodes/Document-impl.js

```diff
@@ -227,6 +227,8 @@ class DocumentImpl extends NodeImpl {
     }
 
     this._contentType = privateData.options.contentType;
+    this._encoding = privateData.options.encoding;
```

Comment:
Hmm, so defaultEncoding is the user-supplied encoding?

Maybe we shouldn't allow user-supplied encodings at all. Browsers generally do not, but instead follow the spec algorithms for sniffing.